sds/src/secondary_validation/aba_rtn_checksum.rs

```diff
@@ -0,0 +1,62 @@
+use crate::secondary_validation::{get_next_digit, Validator};
+
+pub struct AbaRtnChecksum;
+
+impl Validator for AbaRtnChecksum {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        // check if (3(d1 + d4 + d7) + 7(d2 + d5 + d8) + (d3 + d6 + d9)) mod 10 == 0
+        if regex_match.len() != 9 {
+            return false;
+        }
+
+        let mut chars = regex_match.chars();
+        let mut sums = [0; 3];
+
+        for i in 0..9 {
+            if let Some(digit) = get_next_digit(&mut chars) {
+                sums[i % 3] += digit;
+            } else {
+                return false;
+            }
+        }
+
+        let checksum = (3 * sums[0] + 7 * sums[1] + sums[2]) % 10;
+        checksum == 0
```

Comment:
You could avoid the array allocation and compute the checksum in-place by first defining a statically allocated array of coefficients:
```
static ABA_RNT_MULTIPLIERS: &'static [u32] = &[3, 7, 1];
```

From there you can do something like
```suggestion
        let mut checksum: u32 = 0;

        for i in 0..9 {
            if let Some(digit) = get_next_digit(&mut chars) {
                checksum += ABA_RNT_MULTIPLIERS[i % 3] * digit;
            } else {
                return false;
            }
        }

        checksum % 10 == 0
```